### PR TITLE
[user-authn] Update info about implementing authentication into a user application

### PR DESCRIPTION
**Pull request by z9r5**
## Description
Update info about implementing authentication into a user application.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: user-authn
type: chore
summary: Update info about implementing authentication into a user application.
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

File: modules/150-user-authn/docs/FAQ_RU.md [hunk 1]
```diff
@@ -2,50 +2,66 @@
...
-  nginx.ingress.kubernetes.io/auth-url: https://my-cool-app-dex-authenticator.my-cool-namespace.svc.{{ домен вашего кластера, например | cluster.local }}/dex-authenticator/auth
-  nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-User,X-Auth-Request-Email
-```
+## Как защитить мое приложение?
 
-{% endraw %}
+Чтобы включить аутентификацию через Dex для вашего приложения, нужно:
+1. Создать custom resource [DexAuthenticator](cr.html#dexauthenticator).
+
+   Создание `DexAuthenticator` в кластере приводит к созданию экземпляра [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy), подключенного к Dex. После появления custom resource `DexAuthenticator` в кластере в указанном namespace'е появятся необходимые Deployment, Service, Ingress, Secret.
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ_RU.md:12]:
```suggestion
Для включения аутентификации через Dex в приложение, выполните следующие шаги:
1. Создайте custom resource [DexAuthenticator](cr.html#dexauthenticator).

   Создание `DexAuthenticator` в кластере приводит к созданию экземпляра [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy), подключенного к Dex. После появления custom resource `DexAuthenticator` в указанном namespace появятся необходимые объекты Deployment, Service, Ingress, Secret.
```

File: modules/150-user-authn/docs/FAQ_RU.md [hunk 2]
```diff
@@ -2,50 +2,66 @@
...
+     - admins
+     # Список адресов и сетей, с которых разрешено проходить аутентификацию.
+     whitelistSourceRanges:
+     - 1.1.1.1/32
+     - 192.168.0.0/24
+   ```
+
+2. Подключить приложение к Dex.
+
+   Для этого нужно добавить в Ingress-ресурс вашего приложения следующие аннотации:
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ_RU.md:48]:
```suggestion
2. Подключите приложение к Dex.

   Для этого добавьте в Ingress-ресурс приложения следующие аннотации:
```

File: modules/150-user-authn/docs/FAQ_RU.md [hunk 3]
```diff
@@ -2,50 +2,66 @@
...
+
+   - `nginx.ingress.kubernetes.io/auth-signin: https://$host/dex-authenticator/sign_in`
+   - `nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-User,X-Auth-Request-Email`
+   - `nginx.ingress.kubernetes.io/auth-url: https://<NAME>-dex-authenticator.<NS>.svc.{{ C_DOMAIN }}/dex-authenticator/auth`, где:
+      - `NAME` — значение параметра `metadata.name` ресурса `DexAuthenticator`; 
+      - `NS` — значение параметра `metadata.namespace` ресурса `DexAuthenticator`;
+      - `C_DOMAIN` — домен кластера (параметр [clusterDomain](../../installing/configuration.html#clusterconfiguration-clusterdomain) ресурса `ClusterConfiguration`).
+
+   Пример аннотаций на Ingress-ресурсе приложения, для подключения его к Dex:
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ_RU.md:57]:
```suggestion
   Ниже представлен пример аннотаций на Ingress-ресурсе приложения, для подключения его к Dex:
```

File: modules/150-user-authn/docs/FAQ_RU.md [hunk 4]
```diff
@@ -54,7 +70,7 @@ annotations:
 * Если нужно ограничить доступ по IP и оставить прохождение аутентификации в Dex, добавьте аннотацию с указанием разрешенных CIDR через запятую:
 
   ```yaml
-  nginx.ingress.kubernetes.io/whitelist-source-range: 192.168.0.0/32,1.1.1.1`
+  nginx.ingress.kubernetes.io/whitelist-source-range: 192.168.0.0/32,1.1.1.1
   ```
 
 * Если вы хотите, чтобы пользователи из указанных сетей были освобождены от прохождения аутентификации в Dex, а пользователи из остальных сетей были обязаны аутентифицироваться в Dex, добавьте следующую аннотацию:
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ_RU.md:76]:
```suggestion
* Если необходимо, чтобы пользователи из указанных сетей освобождались от прохождения аутентификации в Dex, а пользователи из остальных сетей обязательно аутентифицировались в Dex, добавьте следующую аннотацию:
```

File: modules/150-user-authn/docs/FAQ_RU.md [hunk 5]
```diff
@@ -124,3 +140,5 @@ annotations:
 ## Как Dex защищен от подбора логина и пароля?
 
 Одному пользователю разрешено только 20 попыток входа. Если лимит был израсходован, еще одна попытка будет добавлена каждые 6 секунд.
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ_RU.md:142]:
```suggestion
Одному пользователю разрешено только 20 попыток входа. Если обозначенный лимит израсходован, одна дополнительная попытка будет добавляться каждые 6 секунд.
```

File: modules/150-user-authn/docs/FAQ_RU.md [hunk 6]
```diff
@@ -54,7 +70,7 @@ annotations:
 * Если нужно ограничить доступ по IP и оставить прохождение аутентификации в Dex, добавьте аннотацию с указанием разрешенных CIDR через запятую:
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ_RU.md:71]:
```suggestion
В DexAuthenticator нет встроенной системы управления разрешением аутентификации на основе IP-адреса пользователя. Вместо этого воспользуйтесь аннотациями для Ingress-ресурсов:
* Если нужно ограничить доступ по IP и оставить прохождение аутентификации в Dex, добавьте аннотацию с указанием разрешенных CIDR через запятую:```

File: modules/150-user-authn/docs/FAQ_RU.md (hunk 1 above)
Comment (MariaMakeeva) [reply modules/150-user-authn/docs/FAQ_RU.md:12]: Ок. Сейчас поправлю)

File: modules/150-user-authn/docs/FAQ.md [hunk 7]
```diff
@@ -27,26 +27,20 @@ spec:
...
   - admins
   whitelistSourceRanges:
-  - 1.1.1.1
+  - 1.1.1.1/32
   - 192.168.0.0/24
 ```
 
-{% endraw %}
-
 After the `DexAuthenticator` custom resource is created in the cluster, Kubernetes will make the necessary deployment, service, ingress, secret in the specified namespace.
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ.md:34]:
```suggestion
After the appearance of the custom resource `DexAuthenticator`, the necessary Deployment, Service, Ingress, Secret objects will appear in the specified namespace.
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

File: modules/150-user-authn/docs/FAQ.md [hunk 8]
```diff
@@ -2,15 +2,15 @@
 title: "The user-authn module: FAQ"
 ---
 
+{% raw %}
+
 ## How to secure my application?
 
 It is possible to hide your application behind Dex authentication by using the `DexAuthenticator` custom resource (CR).
 In fact, by creating the DexAuthenticator in a cluster, user creates an instance [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy), which is already connected to Dex.
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ.md]: To enable Dex authentication for an application, follow these steps:
1. Create a custom resource [DexAuthenticator](cr.html#dexauthenticator) ?

Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ.md]:
```suggestion
To enable Dex authentication for an application, follow these steps:
1. Create a custom resource [DexAuthenticator](cr.html#dexauthenticator).
```

File: modules/150-user-authn/docs/FAQ.md [hunk 9]
```diff
@@ -27,26 +27,20 @@ spec:
...
 Add the following annotations to your app's Ingress resource to connect your application to dex:
 
-{% raw %}
-
 ```yaml
 annotations:
   nginx.ingress.kubernetes.io/auth-signin: https://$host/dex-authenticator/sign_in
   nginx.ingress.kubernetes.io/auth-url: https://my-cool-app-dex-authenticator.my-cool-namespace.svc.{{ cluster domain, e.g., | cluster.local }}/dex-authenticator/auth
   nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-User,X-Auth-Request-Email,Authorization
 ```
```
Comment (MariaMakeeva) [inline modules/150-user-authn/docs/FAQ.md]: 2. Connect the app to Dex.

   To do this, add the following annotations to the Ingress resource of the application:

**Review by kirkonru — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

File: modules/150-user-authn/docs/FAQ.md [hunk 10]
```diff
@@ -2,50 +2,66 @@
...
+   metadata:
+     # Dex authenticator pod name prefix.
+     # For example, if the name prefix is `app-name`, then Dex authenticator pods will look like `app-name-dex-authenticator-7f698684c8-c5cjg`.
+     name: app-name
+     # Namespace to deploy Dex authenticator to.
+     namespace: app-ns
+   spec:
+     # Your application's domain. Requests to it will be redirected for Dex authentication.
+     applicationDomain: "app-name.kube.my-domain.com"
+     # A parameter that determines whether to send the `Authorization: Bearer` header to the application. This one is useful in combination with auth_request in NGINX.
```
Comment (z9r5) [inline modules/150-user-authn/docs/FAQ.md:28]:
```suggestion
     # A parameter that determines whether to send the `Authorization: Bearer` header to the application.
     # This one is useful in combination with auth_request in NGINX.
```

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [7477](https://github.com/deckhouse/deckhouse/pull/7477) to the branch [release-1.57](https://github.com/deckhouse/deckhouse/tree/release-1.57) successful!

**Pull request merged**